.github/workflows/tests.yml

```diff
@@ -16,17 +16,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - name: Install Python
-        uses: actions/setup-python@v5
-        with:
-          python-version: "3.11"
-      - name: Install dependencies
-        run: python -m pip install -r requirements/test.txt
```

Comment:
Where do the dependencies get installed now? We need to make sure they are pinned so that dependabot is actually being used to control them.

Reply:
This is only for the style job, so it only requires ruff itself

Reply:
But this will automatically install the latest version of ruff, right? We want to pin that so that PRs don't break just because a new version of ruff comes out.

Reply:
got it